zerver/tests/test_message_flags.py

```diff
@@ -424,6 +436,59 @@ def test_mark_all_in_stream_topic_read(self) -> None:
             if msg.user_profile_id == user_profile.id:
                 self.assertFalse(msg.flags.read)
 
+    def test_mark_all_in_stream_topic_unread(self) -> None:
+        self.login("hamlet")
+        user_profile = self.example_user("hamlet")
+        self.subscribe(user_profile, "test_stream")
+
+        message_id = self.send_stream_message(
+            self.example_user("hamlet"), "test_stream", "hello", "test_topic"
+        )
+        unrelated_message_id = self.send_stream_message(
+            self.example_user("hamlet"), "Denmark", "hello", "Denmark2"
+        )
+        with self.capture_send_event_calls(expected_num_events=1) as events:
+            result = self.client_post(
+                "/json/mark_topic_as_unread",
+                {
+                    "stream_id": get_stream("test_stream", user_profile.realm).id,
+                    "topic_name": "test_topic",
+                },
+            )
+
+        self.assert_json_success(result)
+
+        event = events[0]["event"]
+        expected = dict(
+            operation="add",
+            messages=[message_id],
+            flag="read",
+            type="update_message_flags",
+            all=False,
+            op="unread_all",
```

Comment:
That's the wrong event. `operation` should be `remove` here, since we want to remove the `read` flag.